### ISSUE TITLE
TypeError: blizzapi_1.default is not a constructor

### ISSUE THREAD

**Issue by AlexZeDim** [lukemnet/blizzapi#236]
I start to implement Nest (which is quite popular nowadays) and using it with `blizzapi`.

The code is fine and pretty simple, if you are familiar with it:

```
import BlizzAPI from 'blizzapi';
// (2) import * as BlizzAPI from 'blizzapi';

@Injectable()
export class AppService {
  private readonly logger = new Logger(
    RealmsService.name, true,
  );

  // (1)
  private BNet: BlizzAPI // (2) BlizzAPI .default

  constuctor () {
    this.test()
  }

  async test() {
        this.BNet = new BlizzAPI({ // (2) BlizzAPI .default
          region: 'eu',
          clientId: id,
          clientSecret: secret,
          accessToken: token
      });
     ...other logic
  }
}

```

Unfortunately, I have found a strange error with `import/export`. 

It doesn't matter, what kind of `import` I will be using:  1.`import BlizzAPI from 'blizzapi';` or 2. `import * as BlizzAPI from 'blizzapi';`  both of them gives the same error, relevant with `constuctor` for `default`.

I am not so good with TS, and especially its `import/export` options. Of course, I already have already figure out the solution to the problem:  [ `Error: *.default is not a constructor`](https://stackoverflow.com/questions/42652423/error-default-is-not-a-constructor)

Also, I have tried dig to the root of the problem and trying to make some changes in `*/dist/index.d.ts` according to SO, but it seems that following any other options, according to IDE doesn't bring a solution:

![image](https://user-images.githubusercontent.com/907696/114268374-6a13d580-9a09-11eb-9302-f93682b6c926.png)

or  `export default class BlizzAPI;`

As I understand it, it requires some small fix within the `export` file. I'll be more than welcome to make a PR, at the end of next week probably, as soon as I discover all the caveats of it.

**Comment by lwojcik** [comment]
Thanks for submitting the issue!

I'll look at this too. I'll let you know when I have updates on that.

I've been struggling with imports / exports since my first steps with TS :angry:

I think it's time to deploy a module bundler, but I'll open a separate PR for that. For now I'll try to patch it quick and dirty.

**Comment by lwojcik** [comment]
Sorry I'm keeping you waiting.

I set up a sample Nest app and I managed to reproduce the issue. My VSCode correctly resolves ``blizzapi`` from npm and code completion kicks in but the app itself complains about reading properties of ``undefined``.

I'm still working on this.

**Comment by AlexZeDim** [comment]
> *Sorry I'm keeping you waiting.
Don't worry about that, I am busy with my work too. ^_^
Anyway, it's working with `tsconfig.json` option, and `BlizzAPI` isn't the only such module. So it's doesn't generate many problems, at least, it's always an option to update the docs and mention such an option.

**Comment by lwojcik** [comment]
Okay, so there is good news and mildly bad news.

Good news is that I added [Rollup](https://rollupjs.org/guide/en/) as a module bundler and figured out a config that seems to be doing the job while remaining simple and readable.

Also, with this issue and my past experiences from other codebases, I'm finally convinced to get rid of the default export. Quoting [authors of rollup-plugin-terser](https://www.npmjs.com/package/rollup-plugin-terser):

> **Why named export?**
> 1. Module is a namespace. Default export often leads to function/component per file dogma and makes code less maintainable.
> 2. Interop with commonjs is broken in many cases or hard to maintain.
> 3. Show me any good language with default exports. It's historical javascriptism.

[v1.3.5-beta.2](https://www.npmjs.com/package/blizzapi/v/1.3.5-beta.2) is a version with named export, i.e. it should work like this:

```js
import { BlizzAPI } from 'blizzapi';

// no longer like this:
// import BlizzAPI from 'blizzapi';
``` 

Mildly bad news here is that the fix will be published as a breaking change + docs will need an update. I'll start working on this within next few days.

**Comment by AlexZeDim** [comment]
I have made a [pull request](https://github.com/lukemnet/blizzapi-docs/pull/91) to the docs part with Typescript and Nest.js page. I also checked the docs repo for any mentions of `import BlizzAPI` (and found nothing) but still won't change the classic `commonjs` part with `require`, since most of example as in plain javascript.

*But I am capable to it, if you are willing to replace it for new syntax*

**Comment by lwojcik** [comment]
Good call, thank you very much!

Nah, commonjs part is fine.

**Comment by lwojcik** [comment]
Released v2.0.0 as the one with named export.
